ui/app/AppLayouts/Wallet/popups/swap/SwapModalAdaptor.qml

```diff
@@ -78,6 +78,10 @@ QObject {
         filters: ValueFilter { roleName: "isTest"; value: root.swapStore.areTestNetworksEnabled }
     }
 
+    readonly property string routeErrorMessage: d.routeErrorMessage
+    readonly property bool isRouteEthBalanceInsufficient: d.isRouteEthBalanceInsufficient
+    readonly property bool isRouteTokenBalanceInsufficient: d.isRouteTokenBalanceInsufficient
+
```

Comment:
if we create a functon, these will not have to be exposes from here into the SwapModal

Reply:
hmm it's not only about the message, we also want to know if we need to show the Buy button (and with `Buy ETH` or `Buy Crypto` label)